ui/src/admin/components/chronograf/Organization.js

```diff
@@ -0,0 +1,139 @@
+import React, {Component, PropTypes} from 'react'
+
+import OnClickOutside from 'react-onclickoutside'
+
+import {DEFAULT_ORG_ID} from 'src/admin/constants/dummyUsers'
+
+class Organization extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      reset: false,
+      isEditing: false,
+      isDeleting: false,
+      workingName: this.props.organization.name,
+    }
+  }
+
+  handleNameClick = () => {
+    this.setState({isEditing: true})
+  }
+
+  handleInputBlur = reset => e => {
+    const {onRename, organization} = this.props
+
+    if (!reset && organization.name !== e.target.value) {
+      onRename(organization, e.target.value)
+    }
+
+    this.setState({reset: false, isEditing: false})
+  }
+
+  handleKeyDown = e => {
+    if (e.key === 'Enter') {
+      this.inputRef.blur()
+    }
+    if (e.key === 'Escape') {
+      this.setState(
+        {reset: true, workingName: this.props.organization.name},
+        () => this.inputRef.blur()
+      )
+    }
+  }
+
+  handleFocus = e => {
+    e.target.select()
+  }
+
+  handleDeleteClick = () => {
+    this.setState({isDeleting: true})
+  }
+
+  handleDismissDeleteConfirmation = () => {
+    this.setState({isDeleting: false})
+  }
+
+  handleDeleteOrg = () => {
+    const {onDelete, organization} = this.props
+    this.setState({isDeleting: false})
```

Comment:
Do you need this if the DOM node is going away?

Reply:
👍